Python/omevv/v1/scripts/autoUpgradeOMEALicense.py

```diff
@@ -0,0 +1,204 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+'''
+Description: autoUpgradeOMEALicense. Python script to Auto upgrade OME advanced+ license to servers with OMEVVP:Compliance>status:UNSUPPORTED and idracLicenseStatus": "UNSUPPORTED"
+
+script_examples:
+
+1. this example will import iDRAC OMEA license for the list of servers mentioned in csv having OMEVVP Compliance>status:UNSUPPORTED and idracLicenseStatus": "UNSUPPORTED"
+    python autoUpgradeOMEALicense.py -csvfilepath C:\\Users\\Administrator\\Desktop\\autoupgradeOMEALicense.csv -retry 2
+
+2. To view the help menu for list of args to be passed:
+    python autoUpgradeOMEALicense.py -h
+
+3. Refer InputSampleFiles>autoUpgradeOMEALicenseSample.csv for the sample csv file
+'''
+
+
+import requests,argparse,sys
+import pandas as pd
+
+from omevv_apis_client.models import Credential
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+from getManagementHostComplianceData import HostManagementComplianceWrapper
+from LicenseImport import LicenseImport
+import warnings
+
+
+warnings.filterwarnings("ignore")
+
+
+class AutoUpgradeOMEALicense:
+    def __init__(self):
+        self.failuremsgdict = {"compliance_failure": [], "licenseimport_failure": [], "inputs_required": []}
+        self.passlist = []
+
+
+    def parse_csv(self, filepath):
+        try:
+            df = pd.read_csv(filepath)
+
+            # To check if the mandatory fields are filled-in
+            self.skipdf = df[["servicetag","ome_ip", "vc_uname", "vc_pwd", "vc_uuid", "idrac_ip", "licensefilename", "shareip", "sharetype","sharename"]].isna().any(axis=1)
+
+            self.failuremsgdict["inputs_required"]=self.failuremsgdict["inputs_required"] + list(df[self.skipdf]["servicetag"].values)
+
+            #Create newdf to include the rows which have the mandatory fields filled-in
+            self.newdf = df[~self.skipdf]
+            self.newdf.fillna('', inplace=True)
+        except Exception as e:
+            print("\n FAIL:Exception occured",e)
+
+
+    # <---- Function to check host compliance --->
+    def hostcompliance_check(self, ome_ip, vc_uname, vc_pwd, vc_uuid):
+        base_url = 'https://{ip}/omevv/GatewayService/v1/'.format(ip=ome_ip)
+        credential = Credential(username=vc_uname, password=vc_pwd)
+        hostmgmtcompliancehelper = HostManagementComplianceWrapper()
+
+        hostmgmtcompliancehelper.create_payload(base_url=base_url, omeIp=ome_ip, vcUsercredential=credential,
+                                                vCenterUUID=vc_uuid, compliance_filter="UNSUPPORTED")
+        print(hostmgmtcompliancehelper)
```

Comment:
what is it getting printed here?

Reply:
It's only priting the object info- I can skip this print statement.